### DISCUSSION TITLE
crashes at the step  "Start stable-diffusion"

### DISCUSSION THREAD

**Discussion by darth2602** [TheLastBen/fast-stable-diffusion#639] [Q&A]
hello, I just started a new version of the notebook on Google Colab (with the mention : If you switch between v1.5 and the v2, you will need to re-run the "Requirements" cell" but not yet the auto download of the 2.0 model

but it crashes at the "Start stable-diffusion" step

Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py", line 17, in <module>
    from modules import sd_samplers, sd_models, localization, sd_vae, extensions, script_loading
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 10, in <module>
    import modules.textual_inversion.textual_inversion
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/textual_inversion/textual_inversion.py", line 13, in <module>
    from modules import shared, devices, sd_hijack, processing, sd_models, images, sd_samplers
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/images.py", line 20, in <module>
    from modules.shared import opts, cmd_opts
ImportError: cannot import name 'opts' from 'modules.shared' (/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py)

what to do? thank you in advance

**Comment by yarabsahel** [comment]
> hello, I just started a new version of the notebook (with the mention : If you switch between v1.5 and the v2, you will need to re-run the "Requirements" cell" but not yet the auto download of the 2.0 model
> 
> but it crashes at the "Start stable-diffusion" step
> 
> Traceback (most recent call last): File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in from modules import shared, devices, sd_samplers, upscaler, extensions, localization File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py", line 17, in from modules import sd_samplers, sd_models, localization, sd_vae, extensions, script_loading File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in from modules import prompt_parser, devices, processing, images File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in import modules.sd_hijack File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 10, in import modules.textual_inversion.textual_inversion File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/textual_inversion/textual_inversion.py", line 13, in from modules import shared, devices, sd_hijack, processing, sd_models, images, sd_samplers File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/images.py", line 20, in from modules.shared import opts, cmd_opts ImportError: cannot import name 'opts' from 'modules.shared' (/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py)
> 
> what to do? thank you in advance

I share the same obstacle with you!

File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/images.py", line 20, in
from modules.shared import opts, cmd_opts
ImportError: cannot import name 'opts' from 'modules.shared' (/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py)

**Comment by TheLastBen** [comment]
use the latest colab : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast_stable_diffusion_AUTOMATIC1111.ipynb

**Reply by yarabsahel** [reply]
> use the latest colab : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast_stable_diffusion_AUTOMATIC1111.ipynb

Yes, it's what I use already.

**Comment by darth2602** [comment]
I also have the latest version, it still crashes, however I noticed that I have 3 folders in my SD folder instead of 2

stablediffusion
stable-diffusion
stable-diffusion-webui

it is perhaps normal (a novelty of this new version of the colab, or not, I do not know

**Reply by yarabsahel** [reply]
I still facing the same error. I removed the sd folder [entirely] and started a fresh script install, but I still, get the same msg. 


 File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/textual_inversion/textual_inversion.py", line 13, in <module>
    from modules import shared, devices, sd_hijack, processing, sd_models, images, sd_samplers
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/images.py", line 20, in <module>
    from modules.shared import opts, cmd_opts
ImportError: cannot import name 'opts' from 'modules.shared' (/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py)

**Comment by darth2602** [comment]
I was in version 2.0 I switched to 1.5, :

Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 31, in <module>
    ldm.modules.attention.BasicTransformerBlock.ATTENTION_MODES["softmax-xformers"] = ldm.modules.attention.CrossAttention
AttributeError: type object 'BasicTransformerBlock' has no attribute 'ATTENTION_MODES'

**Comment by MysticSpiderman** [comment]
Following to stay updated on solutions.

**Comment by xaviersingy** [comment]
Same issue.

Started the colab from scratch - I have a Colab Pro subscription:

Renamed my Google Drive folder "sd" into "sd_old"
Restarted the Colab environment to start clean.
Refreshed the colab page (shift-F5) - all settings were empty

Ran every step of the colab:
Model_version: 1.5
entered my token

It ended at the last step with:

Downloading: "https://github.com/DagnyT/hardnet/raw/master/pretrained/train_liberty_with_aug/checkpoint_liberty_with_aug.pth" to /root/.cache/torch/hub/checkpoints/checkpoint_liberty_with_aug.pth
100% 5.10M/5.10M [00:00<00:00, 157MB/s]
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 31, in <module>
    ldm.modules.attention.BasicTransformerBlock.ATTENTION_MODES["softmax-xformers"] = ldm.modules.attention.CrossAttention
AttributeError: type object 'BasicTransformerBlock' has no attribute 'ATTENTION_MODES'


If that is of any relevance, here are the GPU details:

!nvidia-smi
Sat Nov 26 15:16:21 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:00:04.0 Off |                    0 |
| N/A   49C    P8     9W /  70W |      0MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

**Comment by TheLastBen** [comment]
A1111 just added the V2 support, so I have to remove the manual support I added.

**Comment by yarabsahel** [comment]
It's fixed now, thanks @TheLastBen 

For anyone who still facing the problem, please use the latest colab, I had to remove the sd folder and start fresh to get it working.
 https://github.com/TheLastBen/fast-stable-diffusion/issues/640#issuecomment-1328073030

**Reply by xaviersingy** [reply]
it works :-)

**Comment by darth2602** [comment]
yes everything works perfectly, but ... I tested version 2.0 and.....I'm going back to 1.5 ....

**Comment by Crescent1967** [comment]
I deleted all previous versions and folders and I am still running into issues at the Start stable-diffusion step. It seems like the RAM in Google Colab starts maxing out with me only receiving the below text, and nothing after (RAM drops back down).  Sorry, I am not super technically inclined... any suggestions? I have been using 1.5 without any problems until recently.

LatentInpaintDiffusion: Running in v-prediction mode
DiffusionWrapper has 865.93 M params.
^C

**Reply by yarabsahel** [reply]
I did have the same problem, but once I turned to Premium GPU, it worked.

**Reply by TheLastBen** [reply]
Don't use premium GPU, it's a waste of units, set the GPU to standard and set the VM shape to High-RAM

**Reply by Crescent1967** [reply]
I have been using the free version of colab since I am not a high capacity user thus far. I don't see a VM shape dropdown anywhere after I select GPU as my runtime option. Is this something that is even available for the free version? Yesterday I also tried going back to V1.5 and ran into the same problem.

Any other ways around this so I can continue to use? I have been using V1.4 and V1.5 for some time and hope I won't have to look elsewhere or switch over to a paid version.

**Comment by nawnie** [comment]
im having a similar but diffrent issue NameError: name 'demo' is not defined

**Reply by darth2602** [reply]
Same for me :

LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Loading weights [a2a802b2] from /content/gdrive/MyDrive/sd/stable-diffusion-webui/models/Stable-diffusion/model.ckpt
Applying cross attention optimization (Doggettx).
Model loaded.
Loaded a total of 0 textual inversion embeddings.
Embeddings: 
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 212, in <module>
    webui()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 159, in webui
    demo.queue(concurrency_count=111500)
NameError: name 'demo' is not defined

**Reply by TheLastBen** [reply]
I merged a pull request that should fix it

**Reply by darth2602** [reply]
sorry ....

Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization, ui_tempdir
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 10, in <module>
    import modules.textual_inversion.textual_inversion
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/textual_inversion/textual_inversion.py", line 13, in <module>
    from modules import shared, devices, sd_hijack, processing, sd_models, images, sd_samplers
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 8, in <module>
    import safetensors.torch
ModuleNotFoundError: No module named 'safetensors'

**Reply by TheLastBen** [reply]
do this : `!pip install safetensors`

**Reply by darth2602** [reply]
i added "!pip install safetensors" at the end of  Requirements

...
xformers()
cleanup()
!pip install safetensors

**Reply by darth2602** [reply]
Solved !  (you're the Boss ^_^ !)

**Reply by brian-tam** [reply]
Hello! I also see NameError: name 'demo' is not defined, did you solve it?

**Reply by TheLastBen** [reply]
use the latest colab to fix it

**Reply by brian-tam** [reply]
I am using it. Thank you for your amazing contribution!

**Comment by 19960705** [comment]
excuse me,I have tried several times to delete the folder according to the above method, but the following error still appears,could someone help me to solve this

Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization, ui_tempdir
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 10, in <module>
    import modules.textual_inversion.textual_inversion
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/textual_inversion/textual_inversion.py", line 13, in <module>
    from modules import shared, devices, sd_hijack, processing, sd_models, images, sd_samplers
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 8, in <module>
    import safetensors.torch
ModuleNotFoundError: No module named 'safetensors'

**Reply by TheLastBen** [reply]
`!pip install safetensors` and use the colab from the link to get the latest updates

**Reply by 19960705** [reply]
solved it,thank you sooooo much

**Comment by eduardodp-pixel** [comment]
If I get this problem what should I do?
-
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 13, in <module>
    from modules import shared, devices, sd_samplers, upscaler, extensions, localization, ui_tempdir
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_samplers.py", line 11, in <module>
    from modules import prompt_parser, devices, processing, images
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/processing.py", line 15, in <module>
    import modules.sd_hijack
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_hijack.py", line 31, in <module>
    ldm.modules.attention.BasicTransformerBlock.ATTENTION_MODES["softmax-xformers"] = ldm.modules.attention.CrossAttention
AttributeError: type object 'BasicTransformerBlock' has no attribute 'ATTENTION_MODES'

**Reply by TheLastBen** [reply]
remove the folder sd from your gdrive (also from trash) and use the latest colab : https://colab.research.google.com/github/TheLastBen/fast-stable-diffusion/blob/main/fast_stable_diffusion_AUTOMATIC1111.ipynb

**Reply by eduardodp-pixel** [reply]
Thank you! your contribution is great!

**Comment by ClaraSanders** [comment]
This happens to me but only when loading models merged with Automatic111 these two last days. Perhaps the repo is fine, but what is wrong is the merging.
